### PR TITLE
update converter for tms

### PR DESCRIPTION
**Pull request by BenjaminRuston**
## Description

Add some additional flexibility into the tropics (or tomorrow[dot]io) microwave sounder

## Issue(s) addressed

Resolves #

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have run the unit tests before creating the PR

### CODE REVIEW

File: src/hdf5/tropics_2ioda.py [hunk 1]
```diff
@@ -335,9 +344,15 @@ def get_WMO_satellite_ID(attrs_shortname):
...
+    elif 'TMS_S3_BRTTL1B' in attrs_shortname:
+        WMO_sat_ID = TOMORROWIO_S3_sat_ID
+    elif 'TMS_S1_BRTTL1B' in attrs_shortname:
+        WMO_sat_ID = TOMORROWIO_S1_sat_ID
+    elif 'TMS' in attrs_shortname and 'BRTTL1B' in attrs_shortname:
+        WMO_sat_ID = TOMORROWIO_GENERIC_sat_ID
     else:
         WMO_sat_ID = -1
-        print("could not determine satellite from filename: %s" % afile)
+        print("could not determine satellite from filename: %s" % attrs_shortname)
```
Comment (BenjaminRuston) [inline src/hdf5/tropics_2ioda.py:355]: this was a bug using a ... glow ball .... global (afile)

File: src/hdf5/tropics_2ioda.py [hunk 2]
```diff
@@ -335,9 +344,15 @@ def get_WMO_satellite_ID(attrs_shortname):
         WMO_sat_ID = TROPICS06_WMO_sat_ID
     elif 'TROPICS07' in attrs_shortname:
         WMO_sat_ID = TROPICS07_WMO_sat_ID
+    elif 'TMS_S3_BRTTL1B' in attrs_shortname:
+        WMO_sat_ID = TOMORROWIO_S3_sat_ID
+    elif 'TMS_S1_BRTTL1B' in attrs_shortname:
+        WMO_sat_ID = TOMORROWIO_S1_sat_ID
+    elif 'TMS' in attrs_shortname and 'BRTTL1B' in attrs_shortname:
+        WMO_sat_ID = TOMORROWIO_GENERIC_sat_ID
```
Comment (BenjaminRuston) [inline src/hdf5/tropics_2ioda.py:352]: prefer to use this, does tomorrow have a single or multiple WMO satellite IDs ... or any at all yet?

File: src/hdf5/tropics_2ioda.py [hunk 3]
```diff
@@ -134,10 +138,15 @@ def get_data_from_files(afile):
     obs_data = init_obs_loc()
 
     f = h5py.File(afile, 'r')
-    software_version = f.attrs['L1b_SW_Ver'].decode("utf-8").split('.')
-    if int(software_version[0]) >= 3:
+    software_version = True
+    if 'L1b_SW_Ver' in f.attrs.keys():
+        software_version = int(f.attrs['L1b_SW_Ver'].decode("utf-8").split('.')[0]) >=3
```
Comment (PatNichols) [inline src/hdf5/tropics_2ioda.py]:
```suggestion
        software_version = int(f.attrs['L1b_SW_Ver'].decode("utf-8").split('.')[0]) >= 3
```

**Comment by PatNichols** [comment]
coding norms failure:
checking ./tropics_2ioda.py
./tropics_2ioda.py:143:87: E225 missing whitespace around operator
I will fix it.

File: src/hdf5/tropics_2ioda.py [hunk 4]
```diff
@@ -42,6 +42,10 @@
 TROPICS05_WMO_sat_ID = 967
 TROPICS06_WMO_sat_ID = 968
 TROPICS07_WMO_sat_ID = 969
+# what to use here
+TOMORROWIO_S3_sat_ID = 333
+TOMORROWIO_S1_sat_ID = 333
+TOMORROWIO_GENERIC_sat_ID = 333
```
Comment (BenjaminRuston) [inline src/hdf5/tropics_2ioda.py]:
```suggestion
TOMORROWIO_S3_sat_ID = 1333
TOMORROWIO_S1_sat_ID = 1333
TOMORROWIO_GENERIC_sat_ID = 1333
```


tomorrow io ,,, does not have a WMO ID for their sounders yet

File: src/hdf5/tropics_2ioda.py [hunk 5]
```diff
@@ -134,10 +138,15 @@ def get_data_from_files(afile):
...
+    software_version = True
+    if 'L1b_SW_Ver' in f.attrs.keys():
+        software_version = int(f.attrs['L1b_SW_Ver'].decode("utf-8").split('.')[0]) >= 3
+    nc_properties = True
+    if '_NCProperties' in f.attrs.keys():
+        nc_properties = 'version=2' in f.attrs['_NCProperties'].decode("utf-8")
+    if software_version or nc_properties:
         obs_data = get_data(f, obs_data)
-    elif int(software_version[0]) == 2:
+    elif int(f.attrs['L1b_SW_Ver'].decode("utf-8").split('.')[0]) == 2:
```
Comment (BenjaminRuston) [inline src/hdf5/tropics_2ioda.py:149]: no update from tomorrow on this, but will use this until they reprocess and see if this attribute gets added, will keep asking

Comment (fabiolrdiniz) [reply src/hdf5/tropics_2ioda.py:149]: It's good to have it in place already.

**Review by PatNichols — APPROVED** [review]

**Review by fabiolrdiniz — APPROVED** [review]
Thanks, @BenjaminRuston!

**Review by fcvdb — APPROVED** [review]
Thanks Ben!

**Pull request merged**